miqi/agent/tools/shell.py

```
@@ -1208,6 +1406,440 @@ def _build_safe_env(
+            if event_emitter is None:
+                return
+            now = time.monotonic()
+            if now - last_progress < _INSTALL_PROGRESS_INTERVAL_SECONDS:
+                return
+            last_progress = now
+            await event_emitter.emit(ExecCommandOutputDeltaEvent(
+                turn_id=turn_id,
+                tool_call_id=tool_call_id,
+                stream=stream_name,
+                delta=(
+                    f"[system install] 安装进行中（已运行 "
```

Ruff (0.16.2)

[warning] String contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF001)